File: Q12 container with most water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    let left = 0;
    let right = height.length - 1;
    let maxArea = 0;
    
    while (left < right) {
        // Calculate the area
        let area = Math.min(height[left], height[right]) * (right - left);
        // Update the maximum area found so far
        maxArea = Math.max(maxArea, area);
        
        // Move the pointers
        if (height[left] < height[right]) {
            left++;
        } else {
            right--;
        }
    }
    
    return maxArea;
};

File: Q12 Container with most water.py
class Solution:
    def maxArea(self, height):
        left = 0
        right = len(height) - 1
        max_area = 0
        
        while left < right:
            # Calculate the area
            area = min(height[left], height[right]) * (right - left)
            # Update the maximum area found so far
            max_area = max(max_area, area)
            
            # Move the pointers
            if height[left] < height[right]:
                left += 1
            else:
                right -= 1
        
        return max_area
